### DISCUSSION TITLE
How to properly integrate socketio into HyperExpress

### DISCUSSION THREAD

**Discussion by Mint-Joy** [kartikk221/hyper-express#254] [Q&A] [answered]
Question into title

**Comment by telemakhos** [comment]
Why would you integrate that mess into a project build on top of [uWebSockets.js](https://github.com/uNetworking/uWebSockets.js)?

**Reply by Mint-Joy** [reply]
clients with Socketio have already been written, I am expanding the functionality of microservices

**Comment by kartikk221** [comment, marked as answer]
So it all comes down to which methods / properties / behaviors socket.io expects from the Request / Response object.
We try to get as close as possible but hyper-express does not have 100% compatibility and who knows which undocumented getters / methods socket.io may be calling internally from Express.js

The best approach would be to drop-in hyper-express in the place of express when creating your socket.io instance then running through all possible test cases / calls to see If something breaks or uncaught errors occur.

From this point onwards, you can clone hyper-express and then add the neccessary properties / methods for compatibility and do this until socket.io works properly.
